lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -196,11 +201,34 @@ def generate(self, module):
                 self.remove()
                 module.fail_json(msg="%s" % to_native(e))
 
+        elif not self.isPublicKeyValid(module):
+            pubkey = module.run_command([module.get_bin_path('ssh-keygen', True), '-yf', self.path])
+            pubkey = pubkey[1].strip('\n')
+            try:
+                self.changed = True
+                with open(self.path + ".pub", "w") as pubkey_f:
+                    pubkey_f.write(pubkey + '\n')
+                os.chmod(self.path + ".pub", stat.S_IWUSR + stat.S_IRUSR + stat.S_IRGRP + stat.S_IROTH)
+            except IOError:
+                module.fail_json(
+                    msg='The public key is missing or does not match the private key. '
+                        'Unable to regenerate the public key.')
+            self.public_key = pubkey
+
+            if self.comment:
+                try:
+                    args = [module.get_bin_path('ssh-keygen',
```

Comment:
Since this also overwrites the private key (which apparently also contains the comment), you need to also include
```.py
                if os.path.exists(self.path) and not os.access(self.path, os.W_OK):
os.chmod(self.path, stat.S_IWUSR + stat.S_IRUSR)
```
from the `if not self.isPrivateKeyValid(module, perms_required=False) or self.force:` branch.

Reply:
Will do.

Reply:
Done.